src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/judgeapproval/ApproveDraftOrdersMidEventHandler.java

```diff
@@ -0,0 +1,140 @@
+                    .value(AnotherHearingRequest.builder()
+                        .whichOrder(buildWhichOrderDynamicList(draftOrdersWrapper))
+                        .build())
+                    .build()
+            ))
+            .build());
+
+        return GenericAboutToStartOrSubmitCallbackResponse.<FinremCaseData>builder().data(finremCaseData).build();
+    }
+
+    /**
+     * Builds a {@link DynamicList} representing the orders available for selection
+     * based on the given {@link DraftOrdersWrapper}. Each order is represented as a
+     * {@link DynamicListElement} containing a code and label.
+     *
+     * <p>The method processes up to 5 judge approvals, where each approval contributes a list item
+     * if the approval is non-null and the associated document filename is not empty.</p>
+     *
+     * <p>For each {@link JudgeApproval}:</p>
+     * <ul>
+     *     <li>The code is prefixed with "draftOrder" or "psa", depending on the document type.</li>
+     *     <li>The filename is used as the label of the list item.</li>
+     * </ul>
+     *
+     * <p>Example:</p>
+     * <pre>
+     * Code: draftOrder_1, Label: OrderDocument1.pdf
+     * Code: psa_2, Label: PensionSharingAnnex1.pdf
+     * </pre>
+     *
+     * @param draftOrdersWrapper the {@link DraftOrdersWrapper} containing the judge approvals.
+     * @return a {@link DynamicList} populated with items representing the orders.
+     */
+    private DynamicList buildWhichOrderDynamicList(DraftOrdersWrapper draftOrdersWrapper) {
+        List<DynamicListElement> listItems = new ArrayList<>();
+
+        for (int i = 1; i <= 5; i++) {
+            JudgeApproval judgeApproval = approveOrderService.resolveJudgeApproval(draftOrdersWrapper, i);
+            if (judgeApproval != null) {
+                String codePrefix = DRAFT_ORDER == judgeApproval.getDocType() ? DRAFT_ORDER.name() : PSA.name();
+                String code = codePrefix + SEPARATOR + i;
+
+                String filename = getDocumentFileName(judgeApproval);
+                if (!StringUtils.isEmpty(filename)) {
+                    listItems.add(DynamicListElement.builder().code(code).label(filename).build());
+                }
+            }
+        }
+
+        return DynamicList.builder().listItems(listItems).build();
+    }
+
+    private String getDocumentFileName(JudgeApproval judgeApproval) {
+        String filename = null;
+        if (JUDGE_NEEDS_TO_MAKE_CHANGES == judgeApproval.getJudgeDecision()) {
+            filename = judgeApproval.getAmendedDocument() != null
+                ? judgeApproval.getAmendedDocument().getDocumentFilename()
+                : "Unknown Filename";
+        } else if (READY_TO_BE_SEALED == judgeApproval.getJudgeDecision()) {
+            filename = judgeApproval.getDocument() != null
+                ? judgeApproval.getDocument().getDocumentFilename()
+                : "Unknown Filename";
+        }
+        return filename;
+    }
+}
```

Comment:
So if there is no document, we show "unknown filename".

If UX decided on this, fine.  If we came up with it, something like "no document" may be clearer?

Reply:
@hamzahtahirhmcts will change it to throw an exception instead of "Unknown Filename".

Reply:
[8c2fc6d](https://github.com/hmcts/finrem-case-orchestration-service/pull/1997/commits/8c2fc6d3a0209464891242710e958f94e93fcfd2)